### PR TITLE
Readme edits

### PR DESCRIPTION
**Pull request by Preeticp**
## What is the purpose of this change? What does it change?
Review of Readme, language, structure and grammatical changes. 

## Was the change discussed in an issue?
NA. The docs team was asked to review the odo doc set

### CODE REVIEW

**Comment by Preeticp** [comment]
@cdrage, @surajnarwade this is the first commit for the readme with basic edits, please review. I have shared a google doc with you with a few questions, once the answers for them are available I will push another commit if required.

File: README.md [hunk 1]
```diff
@@ -30,7 +38,7 @@ curl -L https://github.com/redhat-developer/odo/raw/master/scripts/install.sh |
 </details>
 
 <details>
-<summary> :package: :apple: MacOS</summary>
+<summary> MacOS</summary>
```
Comment (cdrage) [inline README.md]: I like the emoji's :+1: I find that having them adds a few more graphics and centers the developers attention.. Especially to a page lacking graphics.

File: README.md [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 - Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP and Python
-- Detect changes to your local code and deploy automatically with `odo watch`
-- List all available components and services from your OpenShift cluster
+- Detects changes to your local code and deploys automatically using `odo watch`
+- Lists all available components and services from your OpenShift cluster
 
 ## Installation
 
-> The only requirement is **OpenShift 3.9.0** and above. A recommended way of testing out and using OpenShift locally is [Minishift](https://github.com/minishift/minishift).
+.Prerequisites:
```
Comment (cdrage) [inline README.md]: You should keep `>` as this messes up the formatting and makes the section look cleaner: https://github.com/redhat-developer/odo/blob/50d93b65edfc3422bdcb348553174453a18cc3ef/README.md

File: README.md [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
+You need **OpenShift 3.9.0** and above.
+*Note:* We recommend using [Minishift](https://github.com/minishift/minishift)to test and use OpenShift locally.
```
Comment (cdrage) [inline README.md]: need to add a space before `to`

Comment (cdrage) [inline README.md]: Saying: We recommend using Minishift

And then saying: to test and use OpenShift locally.

Is saying the same thing.. Just saying "We recommend using Minishift" seems to convey exactly what we mean.

File: README.md [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
-- Detect changes to your local code and deploy automatically with `odo watch`
-- List all available components and services from your OpenShift cluster
+- Detects changes to your local code and deploys automatically using `odo watch`
+- Lists all available components and services from your OpenShift cluster
 
 ## Installation
 
-> The only requirement is **OpenShift 3.9.0** and above. A recommended way of testing out and using OpenShift locally is [Minishift](https://github.com/minishift/minishift).
+.Prerequisites:
+You need **OpenShift 3.9.0** and above.
```
Comment (cdrage) [inline README.md]: you `require` instead of `need` maybe?

File: README.md [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
+# Odo - Developer focused CLI for OpenShift
 
 [![Build Status](https://travis-ci.org/redhat-developer/odo.svg?branch=master)](https://travis-ci.org/redhat-developer/odo) [![codecov](https://codecov.io/gh/redhat-developer/odo/branch/master/graph/badge.svg)](https://codecov.io/gh/redhat-developer/odo) [![CircleCI](https://circleci.com/gh/redhat-developer/odo/tree/master.svg?style=svg)](https://circleci.com/gh/redhat-developer/odo/tree/master) [![mattermost](/docs/img/mattermost.svg)](https://chat.openshift.io/developers/channels/odo)
 
-A fast iterative tool for deploying your source code straight to OpenShift.
+OpenShift Do (odo) is a  fast iterative CLI tool for developers who write, build, and deploy applications on OpenShift.
+Existing tools such as `oc` are more operations-focused and require a deep-understanding of Kubernetes and OpenShift concepts.
 
-## Features
+odo is  an opinionated CLI tool that supports fast, iterative development and that abstracts away complex Kubernetes and OpenShift concepts, thus allowing developers to focus on what's most important to them: code.
```
Comment (cdrage) [inline README.md]: We have had a discussion at the face-to-face, but when odo is used as a sentence it's lower-case. But when odo is used at the beginning, it's upper-case.

Comment (cdrage) [reply README.md]: This is a run-off sentence and has too many `and's`, needs to be reworded: `odo is  an opinionated CLI tool that supports fast, iterative development and that abstracts away complex Kubernetes and OpenShift concepts, thus allowing developers to focus on what's most important to them: code.`

`and that abstracts` sounds odd.

There's too much filler in this sentence.

File: README.md [hunk 6]
```diff
@@ -1,27 +1,35 @@
 <img src="./docs/img/openshift.png" width="180" align="right">
 
-# Odo - Developer Focused CLI for OpenShift
+# Odo - Developer focused CLI for OpenShift
 
 [![Build Status](https://travis-ci.org/redhat-developer/odo.svg?branch=master)](https://travis-ci.org/redhat-developer/odo) [![codecov](https://codecov.io/gh/redhat-developer/odo/branch/master/graph/badge.svg)](https://codecov.io/gh/redhat-developer/odo) [![CircleCI](https://circleci.com/gh/redhat-developer/odo/tree/master.svg?style=svg)](https://circleci.com/gh/redhat-developer/odo/tree/master) [![mattermost](/docs/img/mattermost.svg)](https://chat.openshift.io/developers/channels/odo)
 
-A fast iterative tool for deploying your source code straight to OpenShift.
+OpenShift Do (odo) is a  fast iterative CLI tool for developers who write, build, and deploy applications on OpenShift.
```
Comment (cdrage) [inline README.md]: there is an extra space after `is a`

File: README.md [hunk 7]
```diff
@@ -1,27 +1,35 @@
...
+Existing tools such as `oc` are more operations-focused and require a deep-understanding of Kubernetes and OpenShift concepts.
 
-## Features
+odo is  an opinionated CLI tool that supports fast, iterative development and that abstracts away complex Kubernetes and OpenShift concepts, thus allowing developers to focus on what's most important to them: code.
+
+odo improves the developer experience with OpenShift. It is designed to be simple and concise so that you can focus on coding rather than how to deploy your application. odo builds and deploys your code to your cluster as soon as you save your changes, giving you instant feedback to validate your changes in real-time. odo's syntax and design is centered around concepts already familiar to developers, such as: project, application and component.
```
Comment (cdrage) [inline README.md]: again this is a lot of filler...

We already now it improves the developer experience as that was said here: `OpenShift Do (odo) is a  fast iterative CLI tool for developers who write, build, and deploy applications on OpenShift.`

Comment (cdrage) [reply README.md]: three of these sentences start with `Odo does X, Odo does that..` it'd be good to mix it up.

Comment (cdrage) [reply README.md]: `odo improves the developer experience with OpenShift. ` could be removed.

Comment (cdrage) [reply README.md]: This sentence here refers to developers as third person: `OpenShift Do (odo) is a  fast iterative CLI tool for developers who write, build, and deploy applications on OpenShift.`

But here: `odo builds and deploys your code to your cluster as soon as you save your changes, giving you instant feedback to validate your changes in real-time.` refers to developers as first-person (using `your`).

The rest of the documentation refers to the developer in third-person and should continue like that.

File: README.md [hunk 8]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
-
-## Purpose
+For a list of other methods such as installing the latest binary on specific OS installations, visit the [installation page](/docs/installation.md).
 
-OpenShift Do (odo) is a CLI tool for developers who are writing, building, and deploying applications on OpenShift. With odo, developers get an opinionated CLI tool that supports fast, iterative development which abstracts away Kubernetes and OpenShift concepts, thus allowing them to focus on what's most important to them: code.
-
-Odo was created to improve the developer experience with OpenShift. Existing tools such as `oc` are more operations-focused and requires a deep-understanding of Kubernetes and OpenShift concepts. Odo is designed to be simple and concise so you may focus on coding rather than how to deploy your application. Since odo can build and deploy your code to your cluster immediately after you save you changes, you benefit from instant feedback and can thus validate your changes in real-time. Odo's syntax and design is centered around concepts already familiar to developers, such as: project, application and component.
-
-## Demo
+## odo overview
```
Comment (cdrage) [inline README.md]: Seems odd... Could we keep it as `Demo`?

File: README.md [hunk 9]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
 3. Add the location of extracted binary to your PATH environment variable by following [this Wiki page](https://github.com/redhat-developer/odo/wiki/Setting-PATH-variable-on-Windows).
 
 </details>
 
-#### For a list of other methods such as installing the latest mastery binary, or specific OS installations, visit the [installation page](/docs/installation.md).
-
-## Purpose
+For a list of other methods such as installing the latest binary on specific OS installations, visit the [installation page](/docs/installation.md).
 
-OpenShift Do (odo) is a CLI tool for developers who are writing, building, and deploying applications on OpenShift. With odo, developers get an opinionated CLI tool that supports fast, iterative development which abstracts away Kubernetes and OpenShift concepts, thus allowing them to focus on what's most important to them: code.
```
Comment (cdrage) [inline README.md]: I don't like how we move this section back above. Makes the entire front section very cluttered...

Compare both side-by-side and you'll see what I mean.

File: README.md (hunk 8 above)
Comment (cdrage) [inline README.md]: This is also not capitalized... It's a header.

File: README.md [hunk 10]
```diff
@@ -1,27 +1,35 @@
 <img src="./docs/img/openshift.png" width="180" align="right">
 
-# Odo - Developer Focused CLI for OpenShift
+# Odo - Developer focused CLI for OpenShift
```
Comment (cdrage) [inline README.md]: It should stay as `Focused`

File: README.md [hunk 11]
```diff
@@ -1,27 +1,35 @@
...
+OpenShift Do (odo) is a  fast iterative CLI tool for developers who write, build, and deploy applications on OpenShift.
+Existing tools such as `oc` are more operations-focused and require a deep-understanding of Kubernetes and OpenShift concepts.
 
-## Features
+odo is  an opinionated CLI tool that supports fast, iterative development and that abstracts away complex Kubernetes and OpenShift concepts, thus allowing developers to focus on what's most important to them: code.
+
+odo improves the developer experience with OpenShift. It is designed to be simple and concise so that you can focus on coding rather than how to deploy your application. odo builds and deploys your code to your cluster as soon as you save your changes, giving you instant feedback to validate your changes in real-time. odo's syntax and design is centered around concepts already familiar to developers, such as: project, application and component.
+
+
+## Key features
```
Comment (cdrage) [inline README.md]: "features" should be capitalized.

File: README.md [hunk 12]
```diff
@@ -1,27 +1,35 @@
...
 
 - Designed for fast, iterative development cycles
-- 100% client based. No server required within your OpenShift cluster for deployment
+- Completely client based. No server is needed within your OpenShift cluster for deployment
```
Comment (cdrage) [inline README.md]: Previous documentation sounded more straight-forward.

File: README.md [hunk 13]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
 </details>
 
 <details>
-<summary> :package: :checkered_flag: Windows</summary>
-
-#### In order to correctly use odo you must add it to your PATH environment variable
```
Comment (cdrage) [inline README.md]: This was incorrectly removed.

File: README.md [hunk 14]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
-2. Extract the file
+2. Extract the file.
 3. Add the location of extracted binary to your PATH environment variable by following [this Wiki page](https://github.com/redhat-developer/odo/wiki/Setting-PATH-variable-on-Windows).
 
 </details>
 
-#### For a list of other methods such as installing the latest mastery binary, or specific OS installations, visit the [installation page](/docs/installation.md).
-
-## Purpose
+For a list of other methods such as installing the latest binary on specific OS installations, visit the [installation page](/docs/installation.md).
```
Comment (cdrage) [inline README.md]: Binary and OS installation are different. This should be "binary, or" as it previously was.

File: README.md [hunk 15]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
-
-Odo was created to improve the developer experience with OpenShift. Existing tools such as `oc` are more operations-focused and requires a deep-understanding of Kubernetes and OpenShift concepts. Odo is designed to be simple and concise so you may focus on coding rather than how to deploy your application. Since odo can build and deploy your code to your cluster immediately after you save you changes, you benefit from instant feedback and can thus validate your changes in real-time. Odo's syntax and design is centered around concepts already familiar to developers, such as: project, application and component.
-
-## Demo
+## odo overview
 
 [![asciicast](https://asciinema.org/a/225717.svg)](https://asciinema.org/a/225717)
 
-## Deploying an Application using odo
+## Deploying an application using odo
```
Comment (cdrage) [inline README.md]: Capitalization should be as it was previously.

File: README.md [hunk 16]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
 
-After you have odo installed, follow these steps to build, push, and deploy a Node.js application using odo. Examples for other supported languages and runtimes can be found [here](https://github.com/redhat-developer/odo/blob/master/docs/examples.md).
-
-```sh
+After you install odo, follow these steps to build, push, and deploy a Node.js application using odo. Examples for other supported languages and runtimes can be found [here](https://github.com/redhat-developer/odo/blob/master/docs/examples.md).
```
Comment (cdrage) [inline README.md]: Change to "After installing odo, follow these steps to build, push, and deploy a Node.js application."

File: README.md [hunk 17]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
 
-# Start a local OpenShift development cluster by running minishift
-$ minishift start
+1. Start a local OpenShift development cluster by running minishift
+````
+$ minishift start`
```
Comment (cdrage) [inline README.md]: Remove `

File: README.md [hunk 18]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
 
-After you have odo installed, follow these steps to build, push, and deploy a Node.js application using odo. Examples for other supported languages and runtimes can be found [here](https://github.com/redhat-developer/odo/blob/master/docs/examples.md).
-
-```sh
+After you install odo, follow these steps to build, push, and deploy a Node.js application using odo. Examples for other supported languages and runtimes can be found [here](https://github.com/redhat-developer/odo/blob/master/docs/examples.md).
 
-# Start a local OpenShift development cluster by running minishift
-$ minishift start
+1. Start a local OpenShift development cluster by running minishift
+````
```
Comment (cdrage) [inline README.md]: This should be ```sh

File: README.md [hunk 19]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
-# Last, we'll create a way to access the application
+````
+7. Create a URL to access the application and visit it to test it.
+````
 $ odo url create
-
-# Test it / visit the URL
 $ curl nodejs-myproject.192.168.42.147.nip.io
-```
+````
```
Comment (cdrage) [inline README.md]: See comments above.

File: README.md [hunk 20]
```diff
@@ -60,70 +68,70 @@ sudo sh -c 'curl -L https://github.com/redhat-developer/odo/releases/download/v0
...
-  - [Detailed Installation Guide](https://github.com/redhat-developer/odo/blob/master/docs/installation.md)
-  - [Odo User Guide](https://github.com/redhat-developer/odo/blob/master/docs/getting-started.md)
-  - [Usage Examples for Other Languages and Runtimes](https://github.com/redhat-developer/odo/blob/master/docs/examples.md)
-  - [Odo CLI Reference](https://github.com/redhat-developer/odo/blob/master/docs/cli-reference.md)
-  - [Development](https://github.com/redhat-developer/odo/blob/master/docs/development.md)
+  - [Detailed installation guide](https://github.com/redhat-developer/odo/blob/master/docs/installation.md)
+  - [Odo getting started guide](https://github.com/redhat-developer/odo/blob/master/docs/getting-started.md)
+  - [Usage examples for other languages and runtimes](https://github.com/redhat-developer/odo/blob/master/docs/examples.md)
+  - [Odo CLI reference](https://github.com/redhat-developer/odo/blob/master/docs/cli-reference.md)
+  - [Development guide](https://github.com/redhat-developer/odo/blob/master/docs/development.md)
```
Comment (cdrage) [inline README.md]: Capitalize all titles above.

File: README.md (hunk 11 above)
Comment (Preeticp) [reply README.md]: As per CCS documentation style guide and recommendations we keep headings in sentence case.

File: README.md (hunk 1 above)
Comment (Preeticp) [reply README.md]: I would encourage you to do away with these graphics for the following reasons:
- Figures or graphics should be used to clarify or complement information. 
- Lists, bulleted or numbered, are meant to attract the attention of the user, I am not sure we need graphics in addition here.
- A user who wants to install odo, would automatically click on the OS he uses. Once we move to Ascii doc, I would actually recommend using Definition list for this, which would make the names of the OS bold, automatically highlighting them for the users.
- We also need to keep our styles in sync with the conventions followed in the OC doc set, so far I haven't seen graphics used in this manner in the OC doc set.

File: README.md (hunk 13 above)
Comment (Preeticp) [reply README.md]: Added as a leading sentence.

File: README.md (hunk 9 above)
Comment (Preeticp) [reply README.md]: We need an introduction explaining why a user would want to use odo before directing him to the installation section. The key features as they were earlier were brief and imho did not give a holistic idea about odo. 
Purpose of odo/what odo does for the user should compel the user to install it. Having it after install section adds no value.
I agree, with the wall of text comment, I have streamlined the section, ptal and let me know if this is ok.

File: README.md (hunk 15 above)
Comment (Preeticp) [reply README.md]: As per docs standards.

File: README.md (hunk 2 above)
Comment (Preeticp) [reply README.md]: I have added a prerequisites heading here, once this is in asciidoc, rest assured the formatting would just be fine

File: README.md (hunk 20 above)
Comment (Preeticp) [reply README.md]: Done

**Comment by Preeticp** [comment]
@cdrage we welcome detailed reviews as it helps in having a better final product. I have made some changes based on your comments, and provided reasons for those which are not as per documentation standards or conventions.

File: README.md (hunk 5 above)
Comment (Preeticp) [reply README.md]: Afaiu, odo is a cli command and when used it should be with backticks. I would suggest that we consistently use `odo` everywhere we use it, or use OpenShift Do everywhere, especially in the text. We could use attributes to render oso as OpenShift Do in the published docs WDYT?

File: README.md (hunk 1 above)
Comment (cdrage) [reply README.md]: okay, that sounds good to me :+1:

File: README.md [hunk 21]
```diff
@@ -4,24 +4,29 @@
...
 
-- Designed for fast, iterative development cycles
-- 100% client based. No server required within your OpenShift cluster for deployment
-- Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP and Python
-- Detect changes to your local code and deploy automatically with `odo watch`
-- List all available components and services from your OpenShift cluster
+## Key features
+OpenShift Do is designed to be simple and concise with the following key features:
+
+- Its Syntax and design are centered around concepts familiar to developers, such as project, application, and component.
```
Comment (cdrage) [inline README.md]: Syntax shouldn't be capitalized.

File: README.md [hunk 22]
```diff
@@ -4,24 +4,29 @@
...
+- It is completely client based. No server is required within the OpenShift cluster for deployment.
+- It supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python.
+- It detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real-time.
+- It lists all available components and services from the OpenShift cluster
```
Comment (cdrage) [inline README.md]: Missing a period

File: README.md [hunk 23]
```diff
@@ -4,24 +4,29 @@
...
-- 100% client based. No server required within your OpenShift cluster for deployment
-- Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP and Python
-- Detect changes to your local code and deploy automatically with `odo watch`
-- List all available components and services from your OpenShift cluster
+## Key features
+OpenShift Do is designed to be simple and concise with the following key features:
+
+- Its Syntax and design are centered around concepts familiar to developers, such as project, application, and component.
+- It is completely client based. No server is required within the OpenShift cluster for deployment.
+- It supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python.
```
Comment (cdrage) [inline README.md]: IMO, It's / its / it is a bit repetitive. Should we remove it all together, does:

`Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python.`

sound better than

`It upports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python.` ?

**Comment by amitkrout** [comment]
Cluster creation itself failed in prow setup - https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/redhat-developer_odo/1462/pull-ci-redhat-developer-odo-master-e2e-core/163

Restarting the core test
/test e2e-core

**Comment by Preeticp** [comment]
@cdrage I have addressed your review comments, is this good to be merged now? I will start off with the getting started doc today.

**Comment by kadel** [comment]
/lgtm
/hold

@cdrage cancel the hold when you are ready

**Comment by kadel** [comment]
We are having some issues with our CI, it is not related to this PR. -> marging manualy

**Comment by Preeticp** [comment]
Thank you @kadel

**Pull request merged**